linera-chain/src/data_types.rs

```diff
@@ -678,7 +753,29 @@ impl CertificateValue {
     }
 }
 
-impl Event {
+impl MessageBundle {
+    pub fn is_skippable(&self) -> bool {
+        self.messages.iter().all(PostedMessage::is_skippable)
+    }
+
+    pub fn is_tracked(&self) -> bool {
+        let mut tracked = false;
+        for posted_message in &self.messages {
+            match posted_message.kind {
+                MessageKind::Simple | MessageKind::Bouncing => {}
+                MessageKind::Protected => return false,
```

Comment:
Question: I think I missed the reasoning for returning here. Is it an optimization or is it that marking one message as `Protected` marks the bundle as "can't be tracked"?

Reply:
Yes, I think we have to use the "strongest" message kind as the bundle kind:

If even one message is protected, the bundle is protected.
If even one message is tracked (but none protected), the bundle is tracked.
Otherwise the bundle is simple.

(If it's bouncing, all of its messages must be bouncing. We should probably refactor that in the future somehow, so that invalid combination can't be represented.)